src/Orleans.Core.Abstractions/IDs/ActivationAddress.cs

```diff
@@ -32,7 +35,7 @@ public static ActivationAddress GetAddress(SiloAddress silo, GrainId grain, Acti
             // Silo part is not mandatory
             if (grain == null) throw new ArgumentNullException("grain");
 
-            return new ActivationAddress(silo, grain, activation);
+            return interner.FindOrCreate((silo, grain, activation), key => new ActivationAddress(key.Item1, key.Item2, key.Item3));
```

Comment:
I would image this change would impact throughput more in applications which has a lot of new activations, since now `NewActivationAddress` requires a look up, instead of a allocation. Not sure whether we have a load test testing this scenario.

Reply:
I imagine this PR will have very little impact. We don't have any specific tests for this, but it might be most apparent in the transaction stress tests